api/client/proto/authservice.proto

```diff
@@ -977,6 +1019,89 @@ message DeleteWindowsDesktopRequest {
     string Name = 1 [ (gogoproto.jsontag) = "name" ];
 }
 
+// ChangePasswordWithTokenRequest defines a request to change one or more of a user's auth
+// credential.
+message ChangePasswordWithTokenRequest {
+    string TokenID = 1;
+    // SecondFactorToken is an otp value.
+    string SecondFactorToken = 2;
+    // Password is user's password.
+    bytes Password = 3;
+    // U2FRegisterResponse is U2F registration challenge response.
+    U2FRegisterResponse U2FRegisterResponse = 4;
+    // DeviceName is the name of the second factor device.
+    string DeviceName = 5;
+}
+
+// ChangePasswordWithTokenResponse is a response for ChangePasswordWithToken.
+message ChangePasswordWithTokenResponse {
+    // WebSession is a user's web sesssion created from successful changing of password.
+    types.WebSessionV2 WebSession = 1;
+    // RecoveryCodes are user's new recovery codes.
+    repeated string RecoveryCodes = 2;
+}
+
+// CreateRecoveryStartTokenRequest defines a request to create a user token for starting account
+// recovery.
+message CreateRecoveryStartTokenRequest {
+    // Username is the requesting user.
+    string Username = 1;
+    // RecoveryCode is one of the user's recovery code.
+    bytes RecoveryCode = 2;
+    // IsRecoverPassword is a flag to indicate if request is to recover password or second factor.
+    bool IsRecoverPassword = 3;
+}
+
+// AuthenticateUserWithRecoveryTokenRequest defines a request to authenticate the user defined in
+// token with either password or a second factor.
+message AuthenticateUserWithRecoveryTokenRequest {
```

Comment:
This seems misnamed to me, given that AuthenticateUserWithRecoveryTokenRequest seems to return another token.

What is the use for the RPC? How is it different from CreateRecoveryStartToken?

Reply:
I renamed this to `CreateAccountRecoveryApprovedToken` with more description provided in the rpc definition